applications/mne_analyze/extensions/dipolefit/dipolefit.cpp

```diff
@@ -140,13 +218,275 @@ QWidget *DipoleFit::getView()
+
+QSharedPointer<QEntity> DipoleFit::create3DEnityTree(QSharedPointer<EcdSetModel> pModel) const
+{
+    QSharedPointer<QEntity> pRootEntity = QSharedPointer<QEntity>::create();
+    pRootEntity->setObjectName(QStringLiteral("DipoleEntityTree"));
+
+    QVector3D pos, to, from;
+    from = QVector3D(0.0, 1.0, 0.0);
+    double norm;
+
+    for(int i = 0; i < pModel->rowCount(); ++i) {
+        QModelIndex dataIndex = pModel->index(i);
+        INVERSELIB::ECD tempEcd = pModel->data(dataIndex, Qt::DisplayRole).value<INVERSELIB::ECD>();
+
+        pos.setX(tempEcd.rd(0));
+        pos.setY(tempEcd.rd(1));
+        pos.setZ(tempEcd.rd(2));
+
+        norm = sqrt(pow(tempEcd.Q(0),2)+pow(tempEcd.Q(1),2)+pow(tempEcd.Q(2),2));
+
+        to.setX(tempEcd.Q(0)/norm);
+        to.setY(tempEcd.Q(1)/norm);
+        to.setZ(tempEcd.Q(2)/norm);
+
+//        qDebug()<<"EcdDataTreeItem::plotDipoles - from" << from;
+//        qDebug()<<"EcdDataTreeItem::plotDipoles - to" << to;
+
+        QQuaternion final = QQuaternion::rotationTo(from, to);
+
+        //Set dipole position and orientation
+        QMatrix4x4 m;
+        m.translate(pos);
+        m.rotate(final);
+
+        Qt3DCore::QTransform *pTransform = new Qt3DCore::QTransform();
+        pTransform->setMatrix(m);
+
+        Qt3DExtras::QPhongMaterial *pMaterial = new Qt3DExtras::QPhongMaterial();
+        pMaterial->setAmbient(QColor(rand()%255, rand()%255, rand()%255));
+
+        Qt3DExtras::QConeMesh *pDipolGeometry = new Qt3DExtras::QConeMesh();
+        pDipolGeometry->setBottomRadius(0.001f);
+        pDipolGeometry->setLength(0.003f);
+
+        Qt3DCore::QEntity *pEntity = new QEntity(pRootEntity.data());
+
+        pEntity->addComponent(pTransform);
+        pEntity->addComponent(pMaterial);
+        pEntity->addComponent(pDipolGeometry);
+    }
+
+    return pRootEntity;
+}
+
+
+//*************************************************************************************************************
+
+void DipoleFit::onBrowseButtonClicked()
+{
+    //Get the path
+    QString filePath = QFileDialog::getOpenFileName(m_pDipoleFitControl,
+                                tr("Open File"),
+                                QDir::currentPath()+"/MNE-sample-data",
+                                tr("fif File(*.fif)"));
+    if(!filePath.isNull()) {
+        m_dipoleSettings.setMeasurementFilePath(filePath);
+
+    }
+}
+
+
+//*************************************************************************************************************
+
+void DipoleFit::onFitButtonClicked()
+{
+    QString sFitName = m_pDipoleFitControl->getFitName();
+    QString sModelPath = ECD_SET_MODEL_DEFAULT_DIR_PATH + sFitName;
+    if(sFitName.isEmpty()) {
+        m_pCommu->publishEvent(STATUS_BAR_MSG, QStringLiteral("Fit name invalid"));
+        return;
+    }
+
+    m_analyzeData->loadEcdSetModel(m_dipoleSettings.getSettings(), sModelPath);
```

Comment:
Maybe add a comment for readability that AnalzyeData will notify the DipoleFit about the model via the newModelAvailable method.